cpp/src/arrow/util/async_generator_test.cc

```diff
@@ -1299,4 +1299,26 @@ TEST(PushGenerator, Stress) {
   }
 }
 
+TEST(SingleFutureGenerator, Basics) {
+  auto fut = Future<TestInt>::Make();
+  auto gen = MakeSingleFutureGenerator(fut);
+  auto collect_fut = CollectAsyncGenerator(gen);
+  AssertNotFinished(collect_fut);
+  fut.MarkFinished(TestInt{42});
+  ASSERT_FINISHES_OK_AND_ASSIGN(auto collected, collect_fut);
+  ASSERT_EQ(collected, std::vector<TestInt>{42});
+  // Generator exhausted
+  collect_fut = CollectAsyncGenerator(gen);
```

Comment:
There is an `AssertGeneratorExhausted` if you wanted to use it in these tests.

Reply:
Ah, ok. I'll think about it next time then (don't want to wait for another round of CI here).